src/dist/mod.rs

```diff
@@ -532,7 +532,6 @@ impl TargetTriple {
                 (b"NetBSD", b"x86_64") => Some("x86_64-unknown-netbsd"),
                 (b"NetBSD", b"i686") => Some("i686-unknown-netbsd"),
                 (b"DragonFly", b"x86_64") => Some("x86_64-unknown-dragonfly"),
-                (b"SunOS", b"i86pc") => Some("x86_64-unknown-illumos"),
```

Comment:
Wouldn't it be simpler if we just put `#[cfg()]`s on match arms instead in this case? I'm a bit afraid to create more "android" blocks; that one block is required because on android it shadows the other arms...